src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -689,8 +689,9 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
     } else {
       // If neither top nor bottom, it means that the overlay
       // is vertically centered on the origin point.
-      const smallestDistanceToViewportEdge =
-          Math.min(viewport.bottom - origin.y, origin.y - viewport.left);
+      const smallestDistanceToViewportEdge = Math.min(
+        viewport.bottom - origin.y + viewport.top, origin.y);
```

Comment:
Why isn't this `Math.min(viewport.bottom - origin.y, origin.y - viewport.top);`?

Also, prefer breaking at the higher syntactic level (like it was before)

Reply:
Because that ends up including the scroll position again which throws off the position on a scrolled page.

Reply:
Could you expand the comment the call that out here?

Reply:
Done. Also fixed up the line break.